### DISCUSSION TITLE
How to attach debuger to a Community Server Connector / Tomcat

### DISCUSSION THREAD

**Discussion by ThierrySudre** [microsoft/vscode-java-debug#1240] [Q&A]
My project is a tomcat webapps application with java servlets on server side and many js scripts on client side. I have migrate recently this project from eclipse to vscode. My project is now a maven project under vscode using Community Server Connector with a Tomcat 9.0.67 server.  By using tomcat server pull down menu I can start or debug the server and it works fine. My problem is to add a launch.json configuration to attach the debugger to this server and have a preLaunchTask able to debug this integrated server. I dont know how to create a task in tasks.json to run a vscode internal command like >Servers: Start Server (Tomcat 9.0.67).
As my project is very large I can't build a war file each time I want to debug, so I use the maven war:inplace feature to move the compiled classes to the project WEB-INF/classes folder, so ideally I would like to have two actions when I start to debug (F5), first is to maven war:inplace the classes and then debug tomcat server.
I tried maven Tomcat7 Plugin but this plugin was made to used external Tomcat server not integrated in Community Server Connector.
![vsc tomcat](https://user-images.githubusercontent.com/47594758/195833456-04fbb41a-29f0-43f5-ba6c-707140fb5317.jpg)

**Comment by testforstephen** [comment]
@Eskibear Can you help provide some suggestions here?

**Reply by Eskibear** [reply]
The request I understand is, run `mvn war:inplace` and `Debug Server` (from community-service-connector extension) together with one-click (or F5).

Maven extension simply assembles the command, so you can simply add an entry in tasks.json as below:
```
        {
            "label": "my pre-launch maven command",
            "type": "shell",
            "command": "mvn war:inplace ......",
            "group": "build"
        },
```

The "Debug Server" command is registered here, and it looks like a command doing some pre-jobs before passing corresponding params to java debugger. I think maintainers of https://github.com/redhat-developer/rsp-server-community should have better understanding about what was happening behind.

**Reply by ThierrySudre** [reply]
thanks a lot I have declare the task as you propose in task.json and include the call of this task as a "preLaunchTask" in my launch.json file.
Task is running fine and war:inplace is executed succesfully. So I got 50% of my problem, I will now contact the rsp-server-community team to know how to start or attach to a rsp server with the launch.json configuration.
